src/Credentials/CredentialProvider.php

```diff
@@ -705,6 +705,12 @@ private static function loadRoleProfile(
         $sourceProfileName = "";
         if (!empty($roleProfile['source_profile'])) {
             $sourceProfileName = $roleProfile['source_profile'];
+            # in ~/.aws/config all the named profile (except 'default') are
+            # prefix with 'profile '
+            if ($filename == (self::getHomeDir() . '/.aws/config') &&
```

Comment:
Can you please remove the condition `$filename == (self::getHomeDir() . '/.aws/config')` since I do no think is needed. You can leave the other condition `$sourceProfileName != 'default'`.